class.phpmailer.php

```diff
@@ -840,6 +840,13 @@ protected function addAnAddress($kind, $address, $name = '')
             }
             return false;
         }
+        if ($this->hasMultiBytes($address) && function_exists('idn_to_ascii')) {
+            // can't use $this->parseAddresses, because it calls validateAddress, which will fail here
+            list($U,$D) = explode('@',$address);
```

Comment:
Like $this->parseAddresses that does `if ($useimap and function_exists('imap_rfc822_parse_adrlist'))`, I would first try to use here the builtin imap_rfc822_parse_adrlist() to extract the domain name, and then only use your own parser with explode().

Also, your parser should first verify that explode('@',$address) has exactly 2 elements. If there are less, it will produce a PHP Notice error for undefined $D. If there are more elements, the e-mail address is not valid…